org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Synchronized.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+import static org.jacoco.core.test.validation.targets.Stubs.ex;
+import static org.jacoco.core.test.validation.targets.Stubs.nop;
+
+import org.jacoco.core.test.validation.targets.Stubs.StubException;
+
+/**
+ * This test target is a synchronized statement.
+ */
+public class Synchronized {
+
+	private static final Object lock = new Object();
+
+	private static void normal() {
+		nop(); // $line-before$
+		synchronized (lock) { // $line-monitorEnter$
+			nop(); // $line-body$
+		} // $line-monitorExit$
+	} // $line-after$
```

Comment:
Such usage seems legitime to me given that implicit `return` is always presented and always assigned to such line as verified by another test. Moreover we already use such pattern in `Target03` and `EnumImplicitMethods`, however in `InterfaceDefaultMethodsTarget` and `InterfaceOnlyDefaultMethodsTarget` we use explicit `return` statements :laughing: And BTW I've just realized that we test "before" and "after" only in case of exceptions, but not in case of control structures :wink: